### PR TITLE
[CI] Do not double checkout macios-adr to keep the credentials.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Doing a second checkout is error prone. One of the reason is that there is no guarantee that any of our other scripts is not going to remove a random temp directory (actually, the API diff make does remove it).

If the checkout directory is removed, the post checkout action from VSTS will fail with something of the kind:

```
[error]An error occurred trying to start process '/usr/local/bin/git'
with working directory
'/Users/builder/azdo/_work/22/a/change-detection/tmp/src/macios-adr'. No
such file or directory
```

The correct way to workaround the issue is to 'mimic' the way VSTS uses git via https.extraheaders (note that this might not work with old versions of git lfs, but should work with versions newer than 2.7.1).

We add a script that will update the configuration of a vsts git repo to use an extra base auth header with the system.token of the pipeline, which should have access to the repo.

I have added some extra docs in the older script for github so that other developers understand the usage and will not make the same mistake with API diff again.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

**Review by borrillis — APPROVED** [review]

**Review by mcumming — APPROVED** [review]

File: tools/devops/automation/scripts/set_vsts_extra_header.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<#
+  .SYNOPSIS
+  Adds extra headers to a vsts repository so that we can perform git operaitons without the need of auth.
```
Comment (stephen-hawley) [inline tools/devops/automation/scripts/set_vsts_extra_header.ps1:3]: operaitions -> operations

File: tools/devops/automation/scripts/clean_git_vsts.ps1 [hunk 2]
```diff
@@ -1,3 +1,21 @@
+<#
+  .SYNOPSIS
+  Updates the maccore origin url to contain the given token, allowing this way the CI to checkout the repo.
+
+  .DESCRIPTION
+  Devops does not use ssh to check out the repositories, it uses https. This is problematic when we are trying to do a
+  git operation in any of our scripts. To fix that we can prepend the github token to the checkout url, this allows all
+  git operations to be performed without extra steps because the PAT is part of the url.
+
+  The url pattern for this to wori is "https://$GithubToken@github.com/xamarin/maccore.git"
```
Comment (stephen-hawley) [inline tools/devops/automation/scripts/clean_git_vsts.ps1:10]: wori -> work

**Review by stephen-hawley — CHANGES_REQUESTED** [review]
typos

**Comment by mandel-macaque** [comment]
Needs to be merge for API diff to work. The real issue as a typo in the variable name.

**Pull request merged**